primitives/src/currency.rs

```diff
@@ -0,0 +1,59 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use sp_runtime::RuntimeDebug;
+use sp_std::{convert::Into, prelude::*};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+pub trait CurrencyInfo {
+    fn name(&self) -> &str;
+    fn symbol(&self) -> &str;
+    fn decimals(&self) -> u32;
+}
+
+macro_rules! impl_currency_id {
+    ($(#[$meta:meta])*
+	$vis:vis enum CurrencyId {
+        $($(#[$vmeta:meta])* $symbol:ident($name:expr, $deci:literal),)*
+    }) => {
+		$(#[$meta])*
+		$vis enum CurrencyId {
+			$($(#[$vmeta])* $symbol,)*
+		}
+
+		impl CurrencyInfo for CurrencyId {
+			fn name(&self) -> &str {
+				match self {
+					$(CurrencyId::$symbol => $name),*
+				}
+			}
+			fn symbol(&self) -> &str {
+				match self {
+					$(CurrencyId::$symbol => stringify!($symbol),)*
+				}
+			}
+			fn decimals(&self) -> u32 {
+				match self {
+					$(CurrencyId::$symbol => $deci,)*
+				}
+			}
+		}
+    }
+}
+
+impl_currency_id! {
+    #[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash))]
+    pub enum CurrencyId {
+        DOT("Polkadot", 10),
```

Comment:
How is the decimal used?

Reply:
@kaichaosun it's currently not used, I'd like to do it in next PR

Reply:
Let's discuss this one in code diff.
Some thoughts from me,
if it's an asset issuance platform, the storage needs to maintain the decimals for a specific asset, [frame-assets pallet](https://github.com/paritytech/substrate/blob/master/frame/assets/src/types.rs#L101),
but if it is the native asset for a substrate chain, the source of truth of the decimals locates in chain_spec.json, if we maintain the decimals here, if dot change its decimal, we need to upgrade at the same time, this can be struggle.